Add icon highlight groups

jlbas opened the pull request:
This PR adds the ability to set icon highlight groups. The main purpose is to allow consistent colors across icons. Since this was previously mentioned in #92, I figured others would also be interested in having this as a built-in option.

To enable the icon highlight groups, `icon_custom_colors` has to be set. Since the majority probably want the default web-devicons colors, this option is `false` by default. This should also avoid messing with existing configurations.

I also added the `Index` highlight groups to the README since they appeared to be missing.

jlbast commented on lua/bufferline/render.lua:
```diff
@@ -169,7 +170,7 @@ local function render(update_names)
         local iconChar, iconHl = get_icon(buffer_name, vim.fn.getbufvar(buffer_number, '&filetype'), status)
         local hlName = is_inactive and 'BufferInactive' or iconHl
-        iconPrefix = hlName and hl(hlName) or namePrefix
+        iconPrefix = has_icon_custom_colors and hl('Buffer' .. status .. 'Icon') or hlName and hl(hlName) or namePrefix
```
Instead of defining highlight groups, there's also the simpler alternative of having the icons follow the `namePrefix` when `icon_custom_colors` is set:
```lua
iconPrefix = has_icon_custom_colors and namePrefix or hlName and hl(hlName) or namePrefix
```
I like the simplicity in this, but others might like having the ability to set the colors individually.

romgrk commented:
LGTM :)

One thing, if `Buffer*Icon` groups are meant to follow `Buffer*` groups by default, can you make them use `highlight default link` instead of defining them directly? You can either re-purpose `hi_all` to accept special arguments, eg `call s:hi_all(['BufferCurrentIcon', { 'link': 'BufferCurrent' }])`, or create a different function eg `call s:hi_link([...])`.

This will avoid that colorschemes that are barbar-aware need an update after this PR.

jlbast commented:
That's a good idea, I hadn't thought of that. To implement this I ended up using your second idea of creating a new function to do the linking.

romgrk commented:
Good :) Last adjustment, can you update the both the README and the docs highlight section to match the changes?

jlbas commented:
Just updated! I also went ahead and added the g:bufferline.icon_custom_colors option to the Settings section in the docs (same description as the README). Hopefully my description will be clear to the reader.

romgrk commented:
LGTM, thanks for the PR :)
